PAteam.AcrobatReaderExtended.Library/Class1.cs

```diff
@@ -335,6 +336,65 @@ public static bool MergePdfFiles(DirectCollection<string> inputFilePaths, string
             return flag;
         }
 
+        [DirectDom("Check PDF File for Password Protection")]
+        [DirectDomMethod("Check {Input File Full Path} for password protection")]
+        [MethodDescription("Checks if the file in the filepath is password protected or not.")]
+        public static bool IsPdfPasswordProtected(string pdfFilePath)
+        {
+            bool isPasswordProtected = false;
+            PdfReader reader = null;
+            try
+            {                
+                if (_log.IsDebugEnabled)
+                {
+                    _log.Debug("Direct.PDFExtended.Library - Checking pdf file: " + pdfFilePath + " for password protection");
+                }
+                if (pdfFilePath != null)
+                {
+                    if(pdfFilePath.EndsWith(".pdf"))
```

Comment:
Maybe try to load file into File object and then check extension?

Reply:
Resolved in next PR, it will use the ValidateInput method that already being used in other places (just with an empty field name).